vehicle/simvehicle.cc

```diff
@@ -1046,10 +1046,18 @@ void vehicle_t::get_screen_offset( int &xoff, int &yoff, const sint16 raster_wid
 	if(  cnv==NULL  ||  cnv==(convoi_t *)1  ||  !cnv->is_reversed()  ) {
 		return;
 	}
+	const int dir = ribi_t::get_dir(get_direction());
 	// Add offset when the vehicle is reversed.
-	const sint32 steps_delta = (VEHICLE_STEPS_PER_TILE / 2 - get_desc()->get_length_in_steps());
-	xoff += (steps_delta*dx) >> 10;
-	yoff += ((steps_delta*dy) >> 10);
+	sint32 steps_delta;
+	steps_delta = raster_width*(VEHICLE_STEPS_PER_TILE / 2 - get_desc()->get_length_in_steps()+env_t::reverse_base_offsets[dir][2]);
+	if(dx && dy) {
+		steps_delta &= 0xFFFFFC00;
+	}
+	else {
+		steps_delta = (steps_delta*diagonal_multiplier)>>10;
+	}
+	xoff += ((steps_delta*dx) >> 10)+tile_raster_scale_x(env_t::reverse_base_offsets[dir][0],raster_width);
+	yoff += ((steps_delta*dy) >> 10)+tile_raster_scale_y(env_t::reverse_base_offsets[dir][1],raster_width);
```

Comment:
既存コードに倣って、演算子の間にはスペースを入れてください。
```suggestion
	xoff += ((steps_delta*dx) >> 10) + tile_raster_scale_x(env_t::reverse_base_offsets[dir][0],raster_width);
	yoff += ((steps_delta*dy) >> 10) + tile_raster_scale_y(env_t::reverse_base_offsets[dir][1],raster_width);
```

https://github.com/aburch/simutrans/blob/master/documentation/coding_styles.txt に一度目を通してみると良いと思います。

Reply:
修正しました